docs/annotations.md

```diff
@@ -0,0 +1,106 @@
+# Annotations
+
+This document was written in November 2024 assuming
+
+* Python 3.10+
+* Pydantic V2 (not the V1 API backdoor)
+* Mypy 1.10+ or similarly-functional Python type-checker
+* Interchange 0.4+
+
+Consider a class to store information about atoms. It needs to store an atomic mass, a partial charge, and a position in 3-D space. The following class would work fine:
+
+```python
+class Atom:
+    mass: float
+    charge: float
+    position: list
+```
+
+but this has some shortcomings:
+
+* What units are assumed for each attribute?
+* What if the position is a `numpy.ndarray` (or `openmm.Vec3`)?
+* Can instances of this class be serialized to disk? And then de-serialized?
+* Are any fields optional?
+
+OpenFF's combined use of [Pint](https://pint.readthedocs.io/en/stable/) and [Pydantic](https://docs.pydantic.dev/latest/) solves these problems. To leverage these tools,
+
+1. Classes should subclass out of `openff.interchange.pydantic._BaseModel`, which is a thin wrapper around `pydantic.BaseModel`
+2. Annotations should use `openff.interchange._annotations._Quantity`, which is a thin wrapper around `openff.units.Quantity`, itself having the same behavior as `pint.Quantity`, or annotations derived thereof.
```

Comment:
```suggestion
2. Annotations should use the [`openff.interchange._annotations._*Quantity`] types, which are thin wrappers around `openff.units.Quantity`, itself having the same behavior as `pint.Quantity`, or annotations derived thereof.

[`openff.interchange._annotations._*Quantity`]: https://github.com/openforcefield/openff-interchange/blob/v0.4.0/openff/interchange/_annotations.py#L127-L191
```

Just to give us an easy way to see what's available. If this were a public API, this would obvs be a link to the appropriate module's API docs.